prowler/providers/aws/services/directoryservice/directoryservice_service.py

```diff
@@ -179,22 +199,27 @@ def __get_snapshot_limits__(self, regional_client):
                     directory.region == regional_client.region
                     and directory.type != DirectoryType.ADConnector
                 ):
-                    get_snapshot_limits_parameters = {"DirectoryId": directory.id}
-                    snapshot_limit = regional_client.get_snapshot_limits(
-                        **get_snapshot_limits_parameters
-                    )
-                    self.directories[directory.id].snapshots_limits = SnapshotLimit(
-                        manual_snapshots_current_count=snapshot_limit["SnapshotLimits"][
-                            "ManualSnapshotsCurrentCount"
-                        ],
-                        manual_snapshots_limit=snapshot_limit["SnapshotLimits"][
-                            "ManualSnapshotsLimit"
-                        ],
-                        manual_snapshots_limit_reached=snapshot_limit["SnapshotLimits"][
-                            "ManualSnapshotsLimitReached"
-                        ],
-                    )
-
+                    try:
+                        get_snapshot_limits_parameters = {"DirectoryId": directory.id}
+                        snapshot_limit = regional_client.get_snapshot_limits(
+                            **get_snapshot_limits_parameters
+                        )
+                        self.directories[directory.id].snapshots_limits = SnapshotLimit(
+                            manual_snapshots_current_count=snapshot_limit[
+                                "SnapshotLimits"
+                            ]["ManualSnapshotsCurrentCount"],
+                            manual_snapshots_limit=snapshot_limit["SnapshotLimits"][
+                                "ManualSnapshotsLimit"
+                            ],
+                            manual_snapshots_limit_reached=snapshot_limit[
+                                "SnapshotLimits"
+                            ]["ManualSnapshotsLimitReached"],
+                        )
+                    except ClientError as error:
+                        logger.warning(
```

Comment:
Are we going to handle all ClientError this way? Should it be?

Reply:
Changed